.github/actions/next-integration-stat/src/index.js

```diff
@@ -2,10 +2,53 @@ const { context, getOctokit } = require("@actions/github");
 const { info, getInput } = require("@actions/core");
 const { default: stripAnsi } = require("strip-ansi");
 const { default: fetch } = require("node-fetch");
+const fs = require("fs");
 
 // A comment marker to identify the comment created by this action.
 const BOT_COMMENT_MARKER = `<!-- __marker__ next.js integration stats __marker__ -->`;
 
+// Download logs for a job in a workflow run by reading redirect url from workflow log response.
+async function fetchJobLogsFromWorkflow(octokit, token, job) {
+  console.log("Checking test results for the job ", job.name);
+
+  // downloadJobLogsForWorkflowRun returns a redirect to the actual logs
+  const jobLogRedirectResponse =
+    await octokit.rest.actions.downloadJobLogsForWorkflowRun({
+      accept: "application/vnd.github+json",
+      ...context.repo,
+      job_id: job.id,
+    });
+
+  // fetch the actual logs
+  const jobLogsResponse = await fetch(jobLogRedirectResponse.url, {
+    headers: {
+      Authorization: `token ${token}`,
+    },
+  });
+
+  if (!jobLogsResponse.ok) {
+    throw new Error(
+      `Failed to get logsUrl, got status ${jobLogsResponse.status}`
+    );
+  }
+
+  // this should be the check_run's raw logs including each line
+  // prefixed with a timestamp in format 2020-03-02T18:42:30.8504261Z
+  const logText = await jobLogsResponse.text();
+  const dateTimeStripped = logText
+    .split("\n")
+    .map((line) => line.substr("2020-03-02T19:39:16.8832288Z ".length));
+
+  const nextjsVersion = dateTimeStripped
+    .find((x) => x.includes("RUNNING NEXTJS VERSION:") && !x.includes("$("))
```

Comment:
nit: extract `"RUNNING NEXTJS VERSION:"` into a const?

Reply:
yup, will do it following pr for the formatting changes.